parquet-column/src/main/java/org/apache/parquet/column/values/bytestreamsplit/ByteStreamSplitValuesReaderForFLBA.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.parquet.column.values.bytestreamsplit;
+
+import org.apache.parquet.io.api.Binary;
+
+public class ByteStreamSplitValuesReaderForFLBA extends ByteStreamSplitValuesReader {
+  private final byte[] valueByteBuffer;
+
+  public ByteStreamSplitValuesReaderForFLBA(int length) {
+    super(length);
+    valueByteBuffer = new byte[length];
+  }
+
+  @Override
+  public Binary readBytes() {
+    gatherElementDataFromStreams(valueByteBuffer);
+    return Binary.fromConstantByteArray(valueByteBuffer);
```

Comment:
Good catch. I should probably also test for this.

Reply:
Ok, this doesn't change anything unless you also materialize the `Binary` using `Binary::getBytes`, because it's only then that the `byte[]` is actually copied.

I think this begs the question: what is the contract for the `Binary` returned by `ValuesReader::readBytes`? Should its contents survive subsequent calls to `readBytes`?

Reply:
An alternative would be to decode all the data eagerly, like `DeltaBinaryPackingValuesReader` does.